src/artemis/devices/oav/oav_centring.py

```diff
@@ -0,0 +1,692 @@
+            new_y = keep_inside_bounds(new_y, -1500, 1500)
+            new_z = keep_inside_bounds(new_z, -1500, 1500)
+
+            run_num += 1
+
+            # Now move loop to cross hair
+            yield from bps.mv(
+                self.oav_goniometer.x,
+                new_x,
+                self.oav_goniometer.y,
+                new_y,
+                self.oav_goniometer.z,
+                new_z,
+            )
+
+        # Omega is happy to move at same time as xyz
+        yield from bps.mv(self.oav_goniometer.omega, best_omega_angle)
+        LOGGER.info("exiting OAVCentre")
+
+    def calculate_beam_distance_in_microns(self, x, y):
+        self._extract_beam_position()
+        y_to_move = y - self.beam_centre_y
+        x_to_move = self.beam_centre_x - x
+        x_microns = int(x_to_move * self.oav_parameters.micronsPerXPixel)
+        y_microns = int(y_to_move * self.oav_parameters.micronsPerYPixel)
+        return (x_microns, y_microns)
+
+    def microns_to_xyz_move(self, h, v, omega):
+        """
+        Converts micron measurements from pixels into to (x, y, z) motor coordinates. For an overview of the
+        coordinate system for I03 see https://github.com/DiamondLightSource/python-artemis/wiki/Gridscan-Coordinate-System.
+
+        This is designed for phase 1 mx, with the hardware located to the right of the beam, and the z axis is
+        perpendicular to the beam and normal to the rotational plane of the omega axis. When the x axis is vertical
+        then the y axis is anti-parallel to the beam direction.
+        By definition, when omega = 0, the z axis will be positive in the vertical direction and a positive omega
+        movement will rotate clockwise when looking at the viewed down x-axis. This is standard in
+        crystallography.
+        """
+        x = -h
+        angle = math.radians(omega)
+        cosine = math.cos(angle)
+        """
+        These calculations are done as though we are looking at the back of
+        the gonio, with the beam coming from the left. They follow the
+        mathematical convention that Z +ve goes right, Y +ve goes vertically
+        up. X +ve is away from the gonio (away from you). This is NOT the
+        standard phase I convention.
+        """
+        sine = math.sin(angle)
+        z = v * sine
+        y = v * cosine
+        return np.array([x, y, z])
+
+
+def keep_inside_bounds(value, lower_bound, upper_bound):
+    """
+    If value is above an upper bound then the upper bound is returned.
+    If value is below a lower bound then the lower bound is returned.
+    If value is within bounds then the value is returned
+
+    Args:
+        value: the value being checked against bounds
+        lower_bound: the lower bound
+        lower_bound: the upper bound
+    """
+    below_lower = value < lower_bound
+    above_upper = value > upper_bound
+
+    return (
+        below_lower * -1500
+        + above_upper * 1500
```

Comment:
Should: These are the same constants as you pass in to the function, should they just use `lower_bound` and `upper_bound`?